fairroot/base/source/FairFileSource.cxx

```diff
@@ -296,7 +294,7 @@ void FairFileSource::SetInTree(TTree* tempTree)
 {
     fInTree = nullptr;
     fInTree = tempTree;
-    fRootFile = static_cast<TFile*>(tempTree->GetCurrentFile());
+    fRootFile.reset(tempTree->GetCurrentFile());
```

Comment:
I am not sure, whether GetCurrentFile returns an owning pointer. I somwhat doubt it. Do we need the SetInTree API in this way?

Reply:
I have now fixed this in a new way of handling this. But I don't like the result. Maybe someone else has a better idea?